buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-						h.AssertPathExists(t, filepath.Join(layersDir, "A", "layer.ignore"))
-					})
-				})
-			})
-		})
-
-		when("building succeeds with a clear env", func() {
-			it.Before(func() {
-				mockEnv.EXPECT().List().Return(append(os.Environ(), "TEST_ENV=cleared"))
-
-				bpTOML.Buildpack.Version = "v1.clear"
-				bpTOML.Dir = filepath.Join(buildpacksDir, "A", "v1.clear")
-				bpTOML.Buildpack.ClearEnv = true
-			})
-
-			it("should not apply user-provided env vars", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Error: %s\n", err)
-				}
-				if s := cmp.Diff(h.Rdfile(t, filepath.Join(appDir, "build-info-A-v1.clear")),
-					"TEST_ENV: cleared\n",
-				); s != "" {
-					t.Fatalf("Unexpected info:\n%s\n", s)
-				}
-			})
-
-			it("should set CNB_BUILDPACK_DIR", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Error: %s\n", err)
-				}
-				if s := cmp.Diff(h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-buildpack-dir-A-v1.clear")),
-					bpTOML.Dir,
-				); s != "" {
-					t.Fatalf("Unexpected CNB_BUILDPACK_DIR:\n%s\n", s)
-				}
-			})
-		})
-
-		when("building fails", func() {
-			it("should error when layer directories cannot be created", func() {
-				h.Mkfile(t, "some-data", filepath.Join(layersDir, "A"))
-				_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-				if _, ok := err.(*os.PathError); !ok {
-					t.Fatalf("Incorrect error: %s\n", err)
-				}
-			})
-
-			it("should error when the provided buildpack plan is invalid", func() {
-				bpPlan := buildpack.Plan{
-					Entries: []buildpack.Require{
-						{
-							Metadata: map[string]interface{}{"a": map[int64]int64{1: 2}}, // map with non-string key type
-						},
-					},
-				}
-				if _, err := bpTOML.Build(bpPlan, config, mockEnv); err == nil {
-					t.Fatal("Expected error.\n")
-				} else if !strings.Contains(err.Error(), "toml") {
-					t.Fatalf("Incorrect error: %s\n", err)
-				}
-			})
-
-			it("should error when the env cannot be found", func() {
-				mockEnv.EXPECT().WithPlatform(platformDir).Return(nil, errors.New("some error"))
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err == nil {
-					t.Fatal("Expected error.\n")
-				} else if !strings.Contains(err.Error(), "some error") {
-					t.Fatalf("Incorrect error: %s\n", err)
-				}
-			})
-
-			it("should error when the command fails", func() {
-				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-				if err := os.RemoveAll(platformDir); err != nil {
-					t.Fatalf("Error: %s\n", err)
-				}
-				_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-				if err, ok := err.(*buildpack.Error); !ok || err.Type != buildpack.ErrTypeBuildpack {
-					t.Fatalf("Incorrect error: %s\n", err)
-				}
-			})
-
-			when("modifying the env fails", func() {
-				var appendErr error
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L320-L363